channeldb/channel.go

```diff
@@ -1589,6 +1605,83 @@ func (c *OpenChannel) isBorked(chanBucket kvdb.RBucket) (bool, error) {
 	return channel.chanStatus != ChanStatusDefault, nil
 }
 
+// MarkShutdownSent updates the channel status to indicate that a shutdown has
+// been initiated. It also persists the delivery script that we will use in
+// the Shutdown message so that we can guarantee that the same delivery script
+// is used if a re-connect happens.
+func (c *OpenChannel) MarkShutdownSent(localDeliveryScript []byte,
+	locallyInitiated bool) error {
+
+	return c.markShutdownSent(localDeliveryScript, locallyInitiated)
+}
+
+// markShutdownSent is a helper function which modifies the channel status of
+// the receiving channel and inserts a delivery script under the delivery script
+// key. It adds a status which indicates the party that initiated the channel
+// close.
+func (c *OpenChannel) markShutdownSent(deliveryScript []byte,
+	locallyInitiated bool) error {
+
+	c.Lock()
+	defer c.Unlock()
```

Comment:
blocking: We shouldn't be locking mutexes in private functions because it makes it such that we can't reuse common logic paths in the module. I'd recommend moving these locking directives to the public function.